boot/bootchain_test.go

```diff
@@ -302,25 +228,52 @@ func (s *bootchainSuite) TestBootChainEqualForResealComplex(c *C) {
 		},
 	}
 	pb := boot.ToPredictableBootChains(bc)
-	// sorted variant
+	// equal with itself
+	eq := boot.PredictableBootChainsEqualForReseal(pb, pb)
+	c.Check(eq, Equals, true, Commentf("not equal with itself\nself: %v", pb))
+
+	// already sorted
 	pbOther := boot.PredictableBootChains{
 		{
 			BrandID:        "mybrand",
 			Model:          "foo",
 			Grade:          "dangerous",
 			ModelSignKeyID: "my-key-id",
 			AssetChain: []boot.BootAsset{
+				{Role: bootloader.RoleRunMode, Name: "loader", Hashes: []string{"c", "d"}},
```

Comment:
can we make some, all of these tests realistic,  RunMode would come generally after recovery?